### ISSUE TITLE
Debugger doesn't break on breakpoints if pubspec contains a builder section

### ISSUE THREAD

**Issue by escamoteur** [flutter/flutter#50198]
## Steps to Reproduce

Add this here to any flutter project and try to break at a breakpoint.

```
builders:
  jaguar_serializer_cli: 
    git:
      url: https://github.com/Jaguar-dart/jaguar_serializer.git
      path: serializer_cli
```

**Expected results:**
It breaks

**Actual results:** 
it doesn't,

I tried it with all channels form current stable to master and it happens on all of them.

I'm now on:

```
[√] Flutter (Channel master, v1.14.7-pre.120, on Microsoft Windows [Version 10.0.18362.592], locale de-DE)
    • Flutter version 1.14.7-pre.120 at C:\Entwicklung\flutter
    • Framework revision 82c00c915c (2 hours ago), 2020-02-05 06:35:49 -0800
    • Engine revision 81dffd1241
    • Dart version 2.8.0 (build 2.8.0-dev.7.0 c8ed304e97)

[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at C:\Users\escam\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.4.3)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.4.29709.97

[√] Android Studio (version 3.5)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 39.0.3
    • Dart plugin version 191.8423
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)

[√] IntelliJ IDEA Community Edition (version 2018.2)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2018.2.5
    • Flutter plugin version 29.1.3
    • Dart plugin version 182.4892.25

[√] VS Code, 64-bit edition (version 1.41.1)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension version 3.8.0

[√] Connected device (4 available)
    • SM F907B   • R3CM90AK4KM • android-arm64  • Android 9 (API 28)
    • Windows    • Windows     • windows-x64    • Microsoft Windows [Version 10.0.18362.592]
    • Chrome     • chrome      • web-javascript • Google Chrome 79.0.3945.130
    • Web Server • web-server  • web-javascript • Flutter Tools

• No issues found!
```

I guess that will be interesting for you  @jonahwilliams 

This was reported on the Dart Code plugin but got somehow lost

https://github.com/Dart-Code/Dart-Code/issues/1976

**Comment by jonahwilliams** [comment]
Right now with flutter generate, I try to support builders which build to cache and not just source. To do so requires the use of a multiroot scheme - this also stops breakpoints from working.

Since there don't seem to be any builders (Besides angular) using build to cache, we could remove this feature and breakpoints would probably work as is.

**Comment by rrousselGit** [comment]
> Since there don't seem to be any builders (Besides angular) using build to cache

What do you mean by "cache" here?

Because if it's about `build_to: source` vs `build_to: cache` inside the `build.yaml` of code-generators, then I'm surprised by that statement.

**Comment by jonahwilliams** [comment]
Can you link to some of those builders? I would imagine if they are not already using the `builders:` key then they must be creating merged output directories.

**Comment by rrousselGit** [comment]
Pretty much all generators that depends on source_gen/SharedPartBuilder

Some popular names:
mobx_codegen, functional_widget, built_value, json_serializable 

There are many more.

**Comment by jonahwilliams** [comment]
Are they using a merged output directory in instructions?

**Comment by rrousselGit** [comment]
I do not know what that means

**Comment by rrousselGit** [comment]
Actually, let me try to answer (feel free to ignore this if I'm off-topic):

They do rely on a temporary directory where all the generated source is located.
The builders don't generate the `.g.dart` directly next to the source.
Instead when using `SharedPartBuilder`, then generate a `.my_library.part` somewhere inside the `.dart_tool/` folder.
Then `SharedPartBuilder` merges all these `*.part` files into a single `.g.dart` next to the source.


So my theory is:
- if the problem is caused by "trying to handle `.dart` files inside the `.dart_tool` folder", then breaking it should be fine since these generators don't output actual Dart files in that folder.
- if the problem is "trying to handle anything that may output in `.dart_tool`", then that may be a problem


I'm not sure what really is the issue though.

**Comment by jonahwilliams** [comment]
Thanks for the information @rrousselGit

The multiroot scheme would only be required if the final dart file was produce in .dart_tool and it sounds like that isn't the case. The SharedPartBuilder will continue working in this case

**Comment by jonahwilliams** [comment]
If you're interesting, the change is pending at https://github.com/flutter/flutter/pull/50284 and could otherwise be tried out

**Comment by jonahwilliams** [comment]
Fixed via  #50284
